src/palace/manager/celery/tasks/reaper.py

```diff
@@ -0,0 +1,239 @@
+    with task.transaction() as session:
+        collection = session.execute(collection_query).scalars().one_or_none()
+        if collection:
+            task.log.info(f"Deleting {collection!r}.")
+            collection.delete()
+
+    with task.session() as session:
+        # Check if there are more collections to delete
+        collections_awaiting_delete = session.execute(
+            select(func.count(Collection.id)).where(
+                Collection.marked_for_deletion == True
+            )
+        ).scalar()
+
+    if collections_awaiting_delete and collections_awaiting_delete > 0:
+        task.log.info(
+            f"{pluralize(collections_awaiting_delete, 'collection')}"
+            f" waiting for delete. Re-queueing the reaper."
+        )
+        raise task.replace(collection_reaper.s())
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def measurement_reaper(task: Task) -> None:
+    """
+    Remove measurements that are not the most recent
+    """
+    deletion_query = delete(Measurement).where(Measurement.is_most_recent == False)
+    with task.transaction() as session:
+        rows_removed = _execute_delete(session, deletion_query)
+    task.log.info(f"Deleted {pluralize(rows_removed, 'outdated measurement')}.")
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def annotation_reaper(task: Task) -> None:
+    """
+    The annotation must have motivation=IDLING, must be at least 60
+    days old (meaning there has been no attempt to read the book
+    for 60 days), and must not be associated with one of the
+    patron's active loans or holds.
+    """
+    cutoff = utc_now() - timedelta(days=60)
+
+    restrictions = []
+    for t in Loan, Hold:
+        active_subquery = (
+            select(Annotation.id)
+            .join(t, t.patron_id == Annotation.patron_id)
+            .join(
+                LicensePool,
+                and_(
+                    LicensePool.id == t.license_pool_id,
+                    LicensePool.identifier_id == Annotation.identifier_id,
+                ),
+            )
+        )
+        restrictions.append(~Annotation.id.in_(active_subquery))
+
+    deletion_query = delete(Annotation).where(
+        Annotation.timestamp < cutoff,
+        Annotation.motivation == Annotation.IDLING,
+        *restrictions,
+    )
+
+    with task.transaction() as session:
+        rows_removed = _execute_delete(session, deletion_query)
+    task.log.info(f"Deleted {pluralize(rows_removed, 'outdated idling annotation')}.")
+
+
+@shared_task(queue=QueueNames.default, bind=True)
+def hold_reaper(task: Task, batch_size: int = 100) -> None:
+    """
+    Remove seemingly abandoned holds from the database.
+    """
+    cutoff = utc_now() - timedelta(days=365)
+    analytics_service = task.services.analytics.analytics()
+    query = (
+        select(Hold)
+        .where(Hold.start < cutoff, or_(Hold.end == None, Hold.end < utc_now()))
+        .order_by(Hold.id)
+        .limit(batch_size)
```

Comment:
This query has been simplified in two ways:
- We aren't filtering out unlimited access and open access license pools, because you can't place holds on these items anyway. 
- We drop the concept of "source of truth" integrations.
  - The only source of truth integration we had was OPDS for distributors, which only supports unlimited access items, so you couldn't place a hold anyway. 

So this query should be effectively equivalent, even though its dropped some of the conditions.